src/test/java/com/cloudant/tests/ViewsTest.java

```diff
@@ -468,6 +468,89 @@ public void queryPageWithStartAndEndKeyUsingNewObjectWithIntegerBooleanStringArr
         assertThat(page.getResultList().size(), is(2));
     }
 
+    /**
+     * Assert that passing a complex key with integer, string, and
+     * boolean objects in query will produce a result with
+     * JSON object values.
+     */
+    @Test
+    public void queryWithComplexStartEndKeyAndJsonObjectValue() {
+        init();
+
+        View view = db.view("example/boolean_creator_created")
+                .startKey(true ,"uuid",1)
+                .endKey(true ,"uuid", 2000);
+
+        List<Object> result = view.query(Object.class);
+
+        assertThat(result.size(), is(2));
+    }
+
+    /**
+     * Assert that passing a complex key with integer, string, and
+     * boolean objects in queryView will produce a result with
+     * JSON object values.
+     */
+    @Test
+    public void queryPageWithComplexStartEndKeyAndJsonObjectValue() {
+        init();
+
+        View view = db.view("example/boolean_creator_created")
+                .startKey(false ,"uuid",1)
+                .endKey(false ,"uuid", 2000);
+
+        Page page = null;
+        try {
+            page = view.queryPage(30, null, Object.class);
+        } catch (Exception e) {
```

Comment:
I don't think it is good for these tests to swallow exceptions. Removing this try/catch will allow the test to fail if an exception is thrown.